### PR TITLE
WIP: Phpstan 2

### PR DESCRIPTION
**Pull request by simPod**
Here is increase to level 2. 

- [x] Was a bit more tricky thant the previous, had to shuffle class hiearchies a bit. 
- [x] Found some bugs
- [ ] My first time working with PHPSpec but in some cases I think PHPUnit is more suitable. I have added an example of PHPUnit test replacing PHPSpecs `it_should_encrypt_data_using_aes128`. I have not found a way how to assert the correct values without bending src API. It stopped working as I removed returned Message from `encryptData()`. Check it out and lemme know WDYT.

### CODE REVIEW

**Comment by simPod** [comment]
@ChadSikorra Hi, will you have a time to look at this? :)

**Comment by ChadSikorra** [comment]
For sure! Been a hectic week or so with work. Planning on looking at it over the weekend.

File: src/FreeDSx/Snmp/Message/Security/SecurityParametersInterface.php [hunk 1]
```diff
@@ -25,4 +25,6 @@ interface SecurityParametersInterface extends ProtocolElementInterface
      * @return int
      */
     public function getSecurityModel() : int;
+
+    public function getUsername() : string;
```
Comment (ChadSikorra) [inline src/FreeDSx/Snmp/Message/Security/SecurityParametersInterface.php]: It's not a given that all SNMP security models will have a username field. The UserSecurityModel does, but other ones do not. So I would not like to enforce it on the interface. Is it possible to do `instanceof` checks for the UserSecurityParameters class instead where applicable?

File: src/FreeDSx/Snmp/Protocol/ClientProtocolHandler.php [hunk 2]
```diff
@@ -209,6 +209,7 @@ protected function performDiscovery(MessageRequestV3 $message, SecurityModelModu
         $id = $this->generateId();
         $this->setPduId($discovery->getRequest(), $id);
         $response = $this->sendRequestGetResponse($discovery);
+        assert($response instanceof MessageResponseV3);
```
Comment (ChadSikorra) [inline src/FreeDSx/Snmp/Protocol/ClientProtocolHandler.php]: Could you throw a SecurityModelException here instead if null? Something like: Expected a discovery response, but none was received.

Comment (simPod) [reply src/FreeDSx/Snmp/Protocol/ClientProtocolHandler.php]: What about now?

File: src/FreeDSx/Snmp/Message/Security/SecurityParametersInterface.php (hunk 1 above)
Comment (simPod) [reply src/FreeDSx/Snmp/Message/Security/SecurityParametersInterface.php]: What about now?

**Comment by ChadSikorra** [comment]
Closing this out as the PhpStan level is now set to max in this library.

**Pull request closed without merging**